app/models/agreement.rb

```diff
@@ -2,4 +2,14 @@
 # "SCS Performance Agreement Form".
 class Agreement < Report
 
+  def approve!(stage, comment)
```

Comment:
We may think about bringing this up to the parent class. But maybe when this is finished and we know that this is exactly what they want.